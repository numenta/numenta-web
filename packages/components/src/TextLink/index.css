/**
 * Numenta Web Platform and Sites source code
 * MIT License (see LICENSE.txt)
 * Copyright © 2005—2017 Numenta <http://numenta.com>
 */

._visited:visited {
  color: #0277bd;
}

.textlink {
  composes: dim from 'tachyons';
  composes: link from 'tachyons';
  composes: underline-hover from 'tachyons';
  composes: _visited;
  color: #0277bd;
  word-wrap: normal;
}
