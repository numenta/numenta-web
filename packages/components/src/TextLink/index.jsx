// Numenta Web Platform and Sites source code
// MIT License (see LICENSE.txt)
// Copyright © 2005—2017 Numenta <http://numenta.com>

import {IndexLink, Link} from 'react-router'
import {prefixLink} from 'gatsby-helpers'
import React from 'react'

import {triggerGAnalyticsEvent} from 'numenta-web-shared-utils/lib/client'

import styles from './index.css'


/**
 * HTML Text Link - React view component.
 */
const TextLink = ({children, onClick, target, to}) => {
  const instrumentOnClick = (event) => {
    // send ga event for asset link/download
    triggerGAnalyticsEvent(event.target.getAttribute('href'))
    if (onClick) return onClick(event)  // augment
    return true
  }
  const attrs = {
    // default internal non-index link
    className: styles.textlink,
    onClick: instrumentOnClick,
    target,
    to: prefixLink(to),
  }
  let Node = Link

  if (to && (
    to.match(/^.+:/) || to.match(/^\/assets\//) || to.match(/\.pdf$/)
  )) {
    // external link (browser location)
    Node = 'a'
    if (to.match(/^.*:/)) {
      attrs.href = to  // = mailto:etc@blah.com
    }
    else if (to.match(/^\/assets\//) || to.match(/\.pdf$/)) {
      attrs.href = prefixLink(to)  // = /assets/etc/
    }
    delete attrs.to
  }
  else if (to === '/') {
    // internal index main/home link
    Node = IndexLink
  }

  return (
    <Node {...attrs}>
      {children}
    </Node>
  )
}

TextLink.propTypes = {
  children: React.PropTypes.node.isRequired,
  onClick: React.PropTypes.func,
  target: React.PropTypes.string,
  to: React.PropTypes.string.isRequired,
}

export default TextLink
