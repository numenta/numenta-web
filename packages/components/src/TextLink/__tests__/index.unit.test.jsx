import React from 'react'
import renderer from 'react-test-renderer'

import TextLink from '../../TextLink'


describe('TextLink React component', () => {

  it('Renders Default correctly', () => {
    const component = renderer.create(
      <TextLink to="http://numenta.com">
        Numenta Homepage
      </TextLink>
    )
    const tree = component.toJSON()
    expect(tree).toMatchSnapshot()
  })

  it('Renders Augmented correctly', () => {
    const component = renderer.create(
      <TextLink
        onClick={() => {}}
        target="_blank"
        to="http://numenta.com"
      >
        Numenta Homepage
      </TextLink>
    )
    const tree = component.toJSON()
    expect(tree).toMatchSnapshot()
  })

})
