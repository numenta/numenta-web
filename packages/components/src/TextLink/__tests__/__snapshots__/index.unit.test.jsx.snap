exports[`TextLink React component Renders Augmented correctly 1`] = `
<a
  className="textlink"
  href="http://numenta.com"
  onClick={[Function]}
  target="_blank">
  Numenta Homepage
</a>
`;

exports[`TextLink React component Renders Default correctly 1`] = `
<a
  className="textlink"
  href="http://numenta.com"
  onClick={[Function]}
  target={undefined}>
  Numenta Homepage
</a>
`;
